backend/LfClassicData/LfClassicMiniLcmApi.cs

```diff
@@ -25,6 +25,9 @@ public IAsyncEnumerable<ComplexFormType> GetComplexFormTypes()
         return Task.FromResult<ComplexFormType?>(null);
     }
 
+    private Dictionary<Guid, PartOfSpeech>? _partsOfSpeechCacheByGuid = null;
+    private Dictionary<string, PartOfSpeech>? _partsOfSpeechCacheByStringKey = null;
```

Comment:
these need to be invalidated when a part of speech is modified, eg: created, updated, or deleted.

Reply:
LfClassicMiniLcmApi only implements IMiniLcmReadApi so there's nowhere in this class that would modify parts of speech. If parts of speech are modified in the LF Classic project, we could find that out by doing another Mongo query, but I don't expect parts of speech to change in LF very often. I suppose I could use an in-memory cache and expire it after 24 hours.

Reply:
oh that's right, I forgot the context, this is fine you're right we don't need to worry about invalidation